docs(migration): add cockpit plugin to dataMigrator (#6717)
camunda/camunda-docs: docs/guides/migrating-from-camunda-7/data-migrator/cockpit-plugin.md

---
id: cockpit-plugin
title: Cockpit plugin
sidebar_label: Cockpit plugin
description: "Web-based interface for viewing information about skipped entities during the migration process."
---

:::warning Experimental feature
The Cockpit plugin is an **experimental** feature and should be used with caution in production environments.
:::

The Cockpit plugin provides a web-based interface for viewing information about skipped entities during the migration process. It integrates with Camunda 7 Cockpit to give you visibility into which process instances, variables, or other entities were skipped during migration and the reasons why.

For more information on Camunda 7 plugins, see the [Camunda 7 documentation](https://docs.camunda.org/manual/latest/webapps/cockpit/extend/plugins/).

## Prerequisites

To use the Cockpit plugin, run the migrator with the following setting:

```yaml
camunda.migrator:

>   save-skip-reason: true

Reviewer: /change This flag should be documented in our properties reference as well.

Author: Added to [config-properties.md](https://github.com/camunda/camunda-docs/pull/6717/commits/a86a5850c9848e9b02e353b553b65c1e1b446937#diff-f86c10e7e7e7a07d2b63fd1a511cacbc145c70443edc445d008d91f4b1fb0775)

```

:::info Required configuration
The Cockpit plugin requires running the migrator with `save-skip-reason` enabled.  
Without this setting, skip reasons will not be saved to the database, and the plugin will have no data to display.
:::

## Installation

1. **Download the latest release** from the [releases page](https://github.com/camunda/camunda-7-to-8-data-migrator/releases) or **build the plugin**:

   ```bash
   cd plugins/cockpit
   mvn clean install

>    ```

Author: Created PR https://github.com/camunda/camunda-7-to-8-data-migrator/pull/255

> 2. **Deploy the plugin** to your Camunda 7 installation by copying the generated JAR file into the Camunda 7 plugins directory.  
>    For example, in the Tomcat distribution, the path is `/camunda-bpm-ee-tomcat-<Camunda7Version>-ee/server/apache-tomcat-<TomcatVersion>/webapps/camunda/WEB-INF/lib/`.

Reviewer: ```suggestion
2. **Deploy the plugin** to your Camunda 7 installation by copying the generated JAR file into the Camunda 7 plugins directory. For example the paths are:
   * For Tomcat: `./camunda-bpm-ee-tomcat-<camunda-7-version>-ee/server/apache-tomcat-<tomcat-version>/webapps/camunda/WEB-INF/lib/`.
   * For Run: `./camunda-bpm-run-ee-<camunda-7-version>-ee/configuration/userlib/`.
```

3. **Inspect skipped entities in Cockpit** once the plugin has been deployed.

## Using the Cockpit plugin

After installation and configuration, the Cockpit plugin provides:

- **Skipped entity overview**: View all entities that were skipped during migration.
- **Detailed skip reasons**: Understand why specific entities were not migrated.
- **Migration status tracking**: Monitor the overall progress and health of your migration.

## Limitations

As an experimental feature, the Cockpit plugin currently has the following limitations:

- **Performance impact**: Saving skip reasons may affect migration performance for large datasets.
- **Storage requirements**: Additional database storage is required to persist skip reason data.
- **Camunda 7 dependency**: A running Camunda 7 instance with Cockpit is required.

> - **Limited customization**: UI customization options are not yet available.

Author: Moved your suggestions into the limitations page

- You can read more about limitations in the [limitations](/guides/migrating-from-camunda-7/data-migrator/limitations.md#cockpit-plugin) page.
  :::

The Cockpit plugin provides a web-based interface for viewing information about skipped and migrated runtime and history data. It integrates with Camunda 7 Cockpit to give you visibility into which runtime process instances or history data (like variables, flow nodes, etc.) were successfully migrated or skipped during migration and the reasons why.

For more information on Camunda 7 plugins, see the [Camunda 7 documentation](https://docs.camunda.org/manual/latest/webapps/cockpit/extend/plugins/).

## Prerequisites

:::warning Database Configuration Requirement
The Cockpit plugin **only works** when the Data Migrator creates the migration schema in the Camunda 7 database (default behavior), or when both Camunda 7 and Camunda 8 databases point to the same database instance.

Since the Cockpit plugin runs within Camunda 7, it can only access databases that Camunda 7 is connected to. For an example, see the [Data Migrator setup example](config-examples.md#data-migrator).

> :::

Reviewer: Looking at the end result, two prerequisite warnings seem to be a bit much. Also, do these database location requirements need so much attention? Can we add this to the list of prerequisites without a banner? 🤔 

<img width="1011" height="734" alt="Image" src="https://github.com/user-attachments/assets/52a61917-ff6e-4b15-9348-e9be77a52728" />
